feat(docs): reorganize color documentation (#3453)
Orange-OpenSource/Orange-Boosted-Bootstrap: site/src/content/docs/foundation/colors.mdx

Reviewer: Not sure that we can do anything to it but the rendering on mobile looks broken (misaligned) compared to the one on desktop:

<img width="422" height="833" alt="Image" src="https://github.com/user-attachments/assets/e6910f98-4415-4603-a005-59f447de5438" />

Reviewer: Me neither tbh 😕

---
title: Colors
description: OUDS Web’s colors reinforces our brand identity and ensures consistency across all websites.
aliases:
  - "/docs/customize/color-palette/"
  - "/docs/foundation/color-palette/"
  - "/docs/foundation/colors/"
toc: true
---

import { getData } from '@libs/data'
import { getConfig } from '@libs/config'
import { allTokens, getTokenValue } from '@libs/sass-variables'

> import { getVersionedDocsPath } from '@libs/path'

Reviewer: None of the import is needed here

This page lists all the colors available in OUDS Web and their usage, organized by their category. All colors are displayed in both light and dark modes for reference. However, they should be applied using their respective utility classes or custom properties to ensure dynamic adaptation to the [color mode]([[docsref:/foundation/color-modes]]).

## Background

These colors are used for whole pages and structurally meaningful sections backgrounds.

These colors can be used with [our background utilities]([[docsref:/utilities/background]]).

<Callout type="info" title="Usage reference">
  Please refer to the [background colors design reference](https://unified-design-system.orange.com/472794e18/p/217ac6-color/b/11c27d) for more information on usage.
</Callout>

<ColorTable
  colorPrefix="ouds/color/bg/"
  tokenRegexp={/\$ouds-color-bg-(?<colorName>.*)-light/}
  getUsage={(colorName) => `.bg-${colorName}`} />

## Surface

These colors should be used as background for elements that are on another background.

> These colors can be used with [our background utilities]([[docsref:/utilities/background]]). Some of these surface colors have an opacity value and are designed to come on top of another color, we explain [how to achieve that here]([[docsref:/utilities/background#multiple-backgrounds]]).

Reviewer: For me every surface even if not opacity is meant to be palced on a background. The wording you proposed is better

> These colors can be used with [our background utilities]([[docsref:/utilities/background]]). Some of these surface colors have an opacity value and are designed to come on top of another color, we explain [how to achieve that here]([[docsref:/utilities/background#multiple-backgrounds]]).

Author: Yes they are, but they will not always be used together on Figma like the "Multiple backgrounds" paragraph tries to help to implement.

<Callout type="info" title="Usage reference">
  Please refer to the [surface colors design reference](https://unified-design-system.orange.com/472794e18/p/217ac6-color/b/11c27d) for more information on usage.
</Callout>

<ColorTable
  colorPrefix="ouds/color/surface/"
  tokenRegexp={/\$ouds-color-surface-((?!status|brand))(?<colorName>.*)-light/}
  getUsage={(colorName) => `.bg-surface-${colorName}`} />

> <ColorTable

Author: Two reasons: if we don't we would display ouds/color/surface/brand-primary instead of ouds/color/surface/brand/primary. Also this is the way it is displayed on Figma (https://www.figma.com/design/Co2t6wHMf4GB9NJVGs2Hes/-OUDS-Lib--Design-tokens?node-id=14339-583)

  colorPrefix="ouds/color/surface/brand/"
  tokenRegexp={/\$ouds-color-surface-brand-(?<colorName>.*)-light/}
  getUsage={(colorName) => `.bg-surface-brand-${colorName}`} />

### Status

> These surface colors are used specifically to indicate a status.

Author: As we refer to the Boackground utilties page for usage I do not think it is necessary

</Callout>

<ColorTable
  colorPrefix="ouds/color/action/"
  tokenRegexp={/\$ouds-color-action((?!-support|-negative))-(?<colorName>.*)-light/} />

<ColorTable
  colorPrefix="ouds/color/action/support/"
  tokenRegexp={/\$ouds-color-action-support-(?<colorName>.*)-light/} />

<ColorTable
  colorPrefix="ouds/color/action/negative/"
  tokenRegexp={/\$ouds-color-action-negative-(?<colorName>.*)-light/} />

## Overlay

These colors are used for overlay elements like dropdown, modals and tooltips to create depth and emphasis.

<Callout type="info" title="Usage reference">
  Please refer to the [overlay colors design reference](https://unified-design-system.orange.com/472794e18/p/217ac6-color/b/11c27d) for more information on usage.
</Callout>

<Callout type="warning">
  These colors should only be used for custom components development, this is why no utilities are provided, please use directly the custom properties in your CSS.
</Callout>

<ColorTable
  colorPrefix="ouds/color/overlay/"
  tokenRegexp={/\$ouds-color-overlay-(?<colorName>.*)-light/} />

## Always

> These colors are the sole invariant colors defined by OUDS Web, they will stay the same on every brand, modes or contexts.

Reviewer: They are not sole since decorative are invariant, but since we don't show them, I don't know what to do

<Callout type="info" title="Usage reference">
  Please refer to the [surface colors design reference](https://unified-design-system.orange.com/472794e18/p/217ac6-color/t/e89157f0cf) for more information on usage.
</Callout>

<ColorTable
  colorPrefix="ouds/color/surface/"
  tokenRegexp={/\$ouds-color-surface-((?!status|brand))(?<colorName>.*)-light/}
  getUsage={(colorName) => `.bg-surface-${colorName}`} />

<ColorTable
  colorPrefix="ouds/color/surface/brand/"
  tokenRegexp={/\$ouds-color-surface-brand-(?<colorName>.*)-light/}
  getUsage={(colorName) => `.bg-surface-brand-${colorName}`} />

### Status

These surface colors are used specifically to indicate a status.

Like other surface colors, they can be used with [our background utilities]([[docsref:/utilities/background]]).

{
  ['positive', 'info', 'warning', 'negative', 'accent'].map((status) => (
    <ColorTable
      colorPrefix={`ouds/color/surface/status/${status}/`}

>       tokenRegexp={new RegExp(`\\$ouds-color-surface-status-${status}-\(\?<colorName>\.\*\)-light`)}

Author: Works fine as is, would not work w/o escaping.
Also RegExp.escape does not seem to work

</Callout>

<ColorTable
  colorPrefix="ouds/color/action/"
  tokenRegexp={/\$ouds-color-action((?!-support|-negative))-(?<colorName>.*)-light/} />

<ColorTable
  colorPrefix="ouds/color/action/support/"
  tokenRegexp={/\$ouds-color-action-support-(?<colorName>.*)-light/} />

<ColorTable
  colorPrefix="ouds/color/action/negative/"
  tokenRegexp={/\$ouds-color-action-negative-(?<colorName>.*)-light/} />

## Border

These colors are used for borders and dividers.

These colors can be used with [our border utilities]([[docsref:/utilities/border#color]]).

<Callout type="info" title="Usage reference">
  Please refer to the [border colors design reference](https://unified-design-system.orange.com/472794e18/p/217ac6-color/t/0e45bd67e9) for more information on usage.
</Callout>

<ColorTable
  colorPrefix="ouds/color/border/"
  tokenRegexp={/\$ouds-color-border((?!-status|-on|-focus))-(?<colorName>.*)-light/}
  getUsage={(colorName) => `.border-${colorName}`} />

<ColorTable
  colorPrefix="ouds/color/border/status/"
  tokenRegexp={/\$ouds-color-border-status-(?<colorName>.*)-light/}
  getUsage={(colorName) => `.border-status-${colorName}`} />

<Callout type="info">

>   The following border colors do not have utilities because they are used only in a Sass mixin to create focus rings

Reviewer: ```suggestion
  The following border colors do not have utilities because they are used only in a Sass mixin to create focus rings.
```

<Callout type="info" title="Usage reference">
  Please refer to the [always colors design reference](https://unified-design-system.orange.com/472794e18/p/217ac6-color/t/25abc650b5) for more information on usage.
</Callout>

<ColorTable
  colorPrefix="ouds/color/always/"
  tokenRegexp={/\$ouds-color-always-(?<colorName>.*)/} />

## Opacity

These colors only define opacities levels, they are used to tweak visibility, hierarchy or emphasis of elements.

These colors can be used with [our opacity utilities]([[docsref:/utilities/opacity]]).

<Callout type="info" title="Usage reference">
  Please refer to the [opacity colors design reference](https://unified-design-system.orange.com/472794e18/p/217ac6-color/t/b4eb059054) for more information on usage.
</Callout>

<ColorTable
  colorPrefix="ouds/opacity/"
  tokenRegexp={/\$ouds-opacity-(?<colorName>.*)/}
  getUsage={(colorName) => `.opacity-${colorName}`}
  getStyle={(token, mode) =>
    mode === 'light'
      ? {opacity: token.lightValue, backgroundColor: 'var(--bs-color-surface-brand-primary)'}

>       : {opacity: token.darkValue, backgroundColor: 'var(--bs-color-surface-brand-primary)'}} />

Reviewer: ```suggestion
<ColorTable
  colorPrefix="ouds/color/opacity/"
  tokenRegexp={/\$ouds-color-opacity-(?<colorName>.*)/}
  getUsage={(colorName) => `.bg-opacity-${colorName}`} />
```

Not sure if the actual opacity tokens should or should not be in this page tbh.
Not sure if the suggestion opacity is background only or if it can be applied on borders and content as well, we might modify our doc accordingly

Author: Yes, I don't know tbh wdyt @vprothais?
The Opacity tokens list on Figma is not in the Colors table so you have a point.
In any case we could move this paragraph elsewhere (and rename the ColorTable component)

Reviewer 2: Hum, good question. I was about to propose to use a checkerboard pattern as a background to avoid interpreting that this helpers provide some kind of orange declination but maybe we should don't have it here at all ? Since they are defined in color family, I think we can keep them here.
Here is an example of background found on stackoverflow (old school), maybre we can find a better one :
```css
    background-position: 0px 0px, 10px 10px;
    background-size: 20px 20px;
    background-image: linear-gradient(45deg, #eee 25%, transparent 25%, transparent 75%, #eee 75%, #eee 100%),linear-gradient(45deg, #eee 25%, white 25%, white 75%, #eee 75%, #eee 100%);
```
WDYT ?

Reviewer 2: I have a weird rendering with a diagonal appearing in Firefox :

<img width="122" height="67" alt="Image" src="https://github.com/user-attachments/assets/dbd1a902-1404-4128-8410-14056639ac4e" />

In the stackoverflow there is a more recent way to do that : https://stackoverflow.com/questions/35361986/css-gradient-checkerboard-pattern

I did some test and we just need to tweak the background position a bit and it works well for us :
```css
background: repeating-conic-gradient(#fff 0% 25%, #000 0% 50%) 0 0/ 20px 20px;
```

</Callout>

<ColorTable
  colorPrefix="ouds/color/always/"
  tokenRegexp={/\$ouds-color-always-(?<colorName>.*)/}
  multiTheme={false} />

## Opacity

These colors only define opacities levels, they are used to tweak visibility, hierarchy or emphasis of elements.

These colors can be used with [our opacity utilities]([[docsref:/utilities/opacity]]).

<Callout type="info" title="Usage reference">
  Please refer to the [opacity colors design reference](https://unified-design-system.orange.com/472794e18/p/217ac6-color/t/b4eb059054) for more information on usage.
</Callout>

export const opacityBackgroundStyle = "repeating-conic-gradient(var(--bs-color-bg-primary) 0% 25%, var(--bs-color-bg-inverse-high) 0% 50%) 0 0/ 20px 20px";

<ColorTable
  colorPrefix="ouds/opacity/"
  tokenRegexp={/\$ouds-opacity-(?<colorName>.*)/}
  getUsage={(colorName) => `.opacity-${colorName}`}
  multiTheme={false}
  getStyle={(token, mode) =>
    mode === 'light'
      ? {
          opacity: token.lightValue,
          background: opacityBackgroundStyle
        }
      : {
          opacity: token.darkValue,
          background: opacityBackgroundStyle
        }} />

## Decorative

> <Callout type="info" title="Coming soon">
>   Decorative colors are not yet implemented in OUDS Web.

Author: ```suggestion
<Callout type="info" title="For design only">
  Decorative colors are used for illustrations design purpose only, they should not be needed in OUDS Web. Please contact us if you need to use it in your project.
```

<Callout type="info" title="Usage reference">
  Please refer to the [always colors design reference](https://unified-design-system.orange.com/472794e18/p/217ac6-color/t/25abc650b5) for more information on usage.
</Callout>

<ColorTable
  colorPrefix="ouds/color/always/"
  tokenRegexp={/\$ouds-color-always-(?<colorName>.*)/} />

## Opacity

These colors only define opacities levels, they are used to tweak visibility, hierarchy or emphasis of elements.

These colors can be used with [our opacity utilities]([[docsref:/utilities/opacity]]).

<Callout type="info" title="Usage reference">
  Please refer to the [opacity colors design reference](https://unified-design-system.orange.com/472794e18/p/217ac6-color/t/b4eb059054) for more information on usage.
</Callout>

<ColorTable
  colorPrefix="ouds/opacity/"
  tokenRegexp={/\$ouds-opacity-(?<colorName>.*)/}
  getUsage={(colorName) => `.opacity-${colorName}`}
  getStyle={(token, mode) =>
    mode === 'light'
      ? {opacity: token.lightValue, backgroundColor: 'var(--bs-color-surface-brand-primary)'}
      : {opacity: token.darkValue, backgroundColor: 'var(--bs-color-surface-brand-primary)'}} />

## Decorative

> <Callout type="info" title="Coming soon">

Reviewer: Are we sure that we'll implement them soon ? Or maybe never ?

Author: Idk, but if we think not or in a very long time we should explain why. What is the reason we might not implement them or only in the far future? @vprothais ?
